### ISSUE TITLE
Avoid logging sensitive cookie values

### ISSUE THREAD

**Issue by ian-fox** [pfernie/cookie_store#29]
Currently all incoming cookies are logged at a debug level in [cookie_store.rs](https://github.com/pfernie/cookie_store/blob/455d48c96cca868b825cda3f12cce89728e2a08e/src/cookie_store.rs#L83). Since things like session tokens are commonly found it cookies, this seems like a footgun where anybody setting `RUST_LOG=debug` or below will end up potentially leaking those in their logs.

Reqwest has the notion of a [sensitive header](https://docs.rs/reqwest/latest/reqwest/header/struct.HeaderValue.html#method.set_sensitive) which doesn't get logged, I wonder if it would make sense to have a method on the cookie store to say "cookies with these names are expected to be sensitive, don't log their values," or perhaps some sort of config option to make logging cookie values opt-in instead of having them always logged when the `RUST_LOG` level is debug or lower.

Happy to implement it and submit a pull request if either of those are designs you're happy with.

**Comment by pfernie** [comment]
Yes, I think changing this behavior would make sense. Generally the utility (IMO) of the log stmt here is simply to debug situations where the store is behaving unexpectedly, and as the store behavior should not depend on the _value_ of the stored cookie(s), changing the logging here to not log the value at all seems reasonable. This could be as simple as changing the logging statement to log `cookie.name()` instead of the entire `cookie`, although other metadata might be useful in debugging (`Expires`/`Max-Age`, etc.) store behavior, so perhaps logging additional fields or all fields _except_ the value would be useful. As you suggest, the current behavior could be restored via an opt-in configuration `log_cookie_values`.

Cookies may also have a `secure` attribute set; while intended to signal such cookies should only be delivered over secure channels, it may make sense to honor that spirit in logging. In cases where the value is present and `true`, changing the logging to only log something like `inserting secure Set-Cookie` without any actual cookie details might be preferable. And again, perhaps make this the default behavior but provide a configuration `log_secure_cookies` or similar to opt-in to the detailed logging.

I'd be happy to accept a PR implementing any or all of above.
